### DISCUSSION TITLE
Getting Type information for native module?

### DISCUSSION THREAD

**Discussion by KokoDoko** [excaliburjs/Excalibur#2366] [Q&A]
I am importing the excalibur library as a native .esm module from https://esm.sh/excalibur. In my code I use `import` like this

```
import { Engine, Actor, CollisionType, ImageSource, Loader, Vector, Physics } from '../library/excalibur.esm.js'

let game = new Engine({...})
let enemy = new Actor({...})

// etc.
```

This works fine, but I don't get the type information that is included when you use the `node_modules` folder. I have tried extracting `index.d.ts` from there, but it links to other files in `node_modules`

Is there a way to get type information without installing the entire `node_modules` folder? It seems a bit overkill?

**Comment by eonarheim** [comment]
@KokoDoko I'll need to look into esm a bit more, our current esm support isn't the best and is still a little experimental unfortunately. I'll make an issue to look into improving the ESM type story.

Issue https://github.com/excaliburjs/Excalibur/issues/2367

@kamranayub Do you have any ideas around ESM and types?

@KokoDoko  What tools are you using with your esm setup so I can reproduce your environment?

**Reply by KokoDoko** [reply]
I am creating my excalibur project using native .es6 modules, without a module bundler. Loading excalibur this way works quite well! 
I put an example here: https://codesandbox.io/s/excaliquarium-wyyh1m

This problem is not specific to Excalibur, I have also encountered it with ThreeJS and PixiJS. You can load them as a native module, but the type information remains hidden in the `node_modules` folder.
